Realm/RLMSyncUser.h

```diff
@@ -145,6 +153,17 @@ NS_ASSUME_NONNULL_BEGIN
                 onCompletion:(RLMUserCompletionBlock)completion
 NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
 
+/**
+ Create a sync configuration instance.
+ 
+ @param url     The unresolved absolute URL to the Realm on the Realm Object Server, e.g.
+ `realm://example.org/~/path/to/realm`. "Unresolved" means the path should
+ contain the wildcard marker `~`, which will automatically be filled in with
+ the user identity by the Realm Object Server.
+ */
+-(RLMRealmConfiguration *)createConfiguration:(NSURL *)url
```

Comment:
The Cocoa name for this is `configurationWithURL:`.

I was expecting this to return a RLMSyncConfiguration as `automaticConfiguration` did. Returning a RLMRealmConfiguration makes modifying the other RLMSyncConfiguration properties very awkward.

Reply:
Ok, I wasn't 100% how this was used. I'll make it return a `RLMSyncConfiguration` instead.